### DISCUSSION TITLE
how to use not() filter from join table?

### DISCUSSION THREAD

**Discussion by utopictown** [supabase/supabase#20091] [Questions]
i want to query table a with filtering table c, a and c are connected through b. how do i query it in supabase? chatgpt gives me something like this

```
await supabaseClient
    .from("a")
    .select(
      "*, b(id, c(startDate, endDate))"
    )
    .not(
      "id",
      "in",
      `(SELECT b.aid
        FROM b
        JOIN c ON b.cid = c.id 
        WHERE c.startDate <= ${startDateValue} AND 
              c.endDate >= ${endDateValue})`
    );
```

 but i got 

```
error {
  code: '22P02',
  details: null,
  hint: null,
  message: 'invalid input syntax for type uuid: "SELECT.....
```

i have read the not() documentation here [supabase not()'s docs](https://supabase.com/docs/reference/javascript/not) but still not sure how to do it for join table

i have checked all the policies and works fine when i query it without not()

thank you in advance

**Comment by GaryAustin1** [comment]
There is no way you can put SQL code in the REST API calls.
If b is a join table with foreign keys to both a and c then as long as you make them a composite primary key,  you can just use c(*) without b(c...) in the query.
The .not filter should then work on .not('c.col','eq','value').  b.c.col should also work.
https://supabase.com/docs/reference/javascript/not
https://supabase.com/docs/reference/javascript/using-filters?example=filter-referenced-tables

**Reply by whoiscarlo** [reply]
What if you're trying to select all of the rows in **Table_A** where `id` is not in any of **Table_B** rows, where **Table_B** has a column `foo` which is a foreign key of **Table_A** column `id`.

Meaning how do you achieve this without making a function: 
```
SELECT 
    Table_A.id, 
FROM 
    Table_A
WHERE 
    AND Table_A.id NOT IN (
        SELECT Table_B.foo FROM  Table_B
    );
```

**Reply by utopictown** [reply]
when i filter using c.col as the column value, the result did not actually filtered, instead it just gives me full result but with foreign table nulled

```
supabaseClient
    .from("b")
    .select("c(*)")
    .eq("c.startDate", "2023-10-23 17:00:00")
    .eq("c.endDate", "2023-10-26 17:00:00");
```
    
  it gives me full list of b data with c: null except for the row that matched to the eq filter

```
  { orders: null },
  { orders: null },  
  { orders: null },
  {
    orders: {
        id: ...
     }
  }
```

**Reply by tomelliot** [reply]
@utopictown I'm seeing the same issue where the foreign table is just `null` (rather than the row not being returned at all). Did you find a solution? This seems like a bug to me.

**Reply by jasonmit** [reply]
@tomelliot I was running into something similar and found this workaround: https://github.com/orgs/supabase/discussions/6587#discussioncomment-2657550
